customer.go

```diff
@@ -7,21 +7,23 @@ import (
 // CustomerParams is the set of parameters that can be used when creating or updating a customer.
 // For more details see https://stripe.com/docs/api#create_customer and https://stripe.com/docs/api#update_customer.
 type CustomerParams struct {
-	Params
-	Balance        int64
-	BalanceZero    bool
-	Token, Coupon  string
-	CouponEmpty    bool
-	Source         *SourceParams
-	Desc, Email    string
-	Plan           string
-	Quantity       uint64
-	TrialEnd       int64
-	DefaultSource  string
-	Shipping       *CustomerShippingDetails
-	BusinessVatID  string
-	TaxPercent     float64
-	TaxPercentZero bool
+	Params         `form:"*"`
+	Balance        int64                    `form:"account_balance"`
+	BalanceZero    bool                     `form:"account_balance,zero"`
```

Comment:
Yes, I think pointers would've been a better decision (although they actually come with their own weird oddities because you can't reference an inline literal like `Balance: &123` so you need to introduce other helper methods), but I don't want to totally break everything for now.

Reply:
Makes sense. I think the solution we have now works fine especially if we make another pass in the future to verify every property once and for all. Was more curious how you would approach it if you were rewriting this entirely today (from scratch). I really like the empty/invert approach you designed.